src/spaceapi_client.rs

```diff
@@ -1,84 +1,203 @@
 //extern crate spaceapi;
+use std::io::prelude::*;
 use rustc_serialize::json;
 use hyper::Client;
-use std::io::Read;
-use spaceapi::Optional::{Value,Absent};
+use spaceapi::Optional::{self, Value, Absent};
 use spaceapi::{Status};
 use spaceapi::sensors::{PeopleNowPresentSensor};
 
-struct SpaceApiClient {
+use std::time::Duration;
+use time::SteadyTime;
+use std::env;
+use std::fs::{self, File};
+use std::io;
+
+pub struct SpaceApiClient {
   last_fetch: i64,
+  webcams: Vec<String>,
+}
+
+impl SpaceApiClient {
+  pub fn new() -> SpaceApiClient {
+    let mut s = SpaceApiClient{
+      last_fetch: 0,
+      webcams: Vec::new(),
+    };
+    
+    s.fetch_webcams();
+    
+    s
+  }
+  
+  pub fn fetch_people_now_present(&self) -> ::std::result::Result<u64, String> {
+    let status = try!(fetch_status());
+    
+    extract_people_now_present(status)
+  }
+
+  pub fn get_webcams(&self) -> Vec<String> {
+    self.webcams.clone()
+  }
```

Comment:
But the user can still do `sac.webcams.clone()`.

If you want to keep it this way, at least I'd call the function `clone_webcams`. Right now it's not really obvious what it does.